src/ngx_http_lua_common.h

```diff
@@ -298,10 +291,13 @@ union ngx_http_lua_srv_conf_u {
 #endif
 
     struct {
-        ngx_http_lua_srv_conf_handler_pt  handler;
-        ngx_str_t           src;
-        u_char             *src_key;
-        int                 src_ref;
+        ngx_http_lua_srv_conf_handler_pt     handler;
+        ngx_str_t                            src;
+        u_char                              *src_key;
+        int                                  src_ref;
```

Comment:
Be careful about struct field memory layout. This `int` creates a 4-byte whole in the middle of this struct.

Reply:
We usually sort the fields by size by convention.

Reply:
I am aware, but this simply follows the layout of the above struct.

Reply:
From my observations, many structs have similar layout flaws, so I assumed we stopped paying attention to those. Easy to update back :+1: